### PR TITLE
core: remove redundant `len` check

### PR DESCRIPTION
**Pull request by Juneezee**
**Description of your changes:**

This pull request is a minor code cleanup.

From the Go specification (https://go.dev/ref/spec#For_range):

> "1. For a nil slice, the number of iterations is 0."
> "3. If the map is nil, the number of iterations is 0."

`len` returns 0 if the slice or map is nil (https://pkg.go.dev/builtin#len). Therefore, checking `len(v) > 0` before a loop is unnecessary.

**Which issue is resolved by this Pull Request:**

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by Juneezee — COMMENTED** [review]
Tip: Hide the whitespace to make the diff easier to read

image

image

File: pkg/operator/ceph/csi/util.go [hunk 1]
```diff
@@ -200,22 +199,20 @@ func applyVolumeToPodSpec(opConfig map[string]string, configName string, podspec
...
+		found := false
+		for j := range podspec.Volumes {
+			// check do we need to override any existing volumes
+			if volumes[i].Name == podspec.Volumes[j].Name {
+				podspec.Volumes[j] = volumes[i]
+				found = true
+				break
 			}
 		}
+		if !found {
```
[the remark this answers is not in the document]
Comment (Juneezee) [reply pkg/operator/ceph/csi/util.go:212]: Hi @parth-gr. Thanks for reviewing this PR.

Not sure if I'm understanding you correctly, but this line only the identation has changed after removing the outermost
```
if len(volumes) > 0 {
}
```

image

**Review by parth-gr — APPROVED** [review]
LGTM, thanks for the cahnges

File: pkg/operator/ceph/test/container.go [hunk 2]
```diff
@@ -76,27 +75,26 @@ func (d *ContainerTestDefinition) TestContainer(
...
 		assert.Equal(t, *d.EnvCount, len(cont.Env))
 	}
-	if d.Ports != nil {
-		assert.Equal(t, len(d.Ports), len(cont.Ports))
-		for i, p := range d.Ports {
-			assert.Equal(t, p.ContainerPort, cont.Ports[i].ContainerPort)
-			assert.Equal(t, p.Protocol, cont.Ports[i].Protocol)
-		}
+
+	assert.Equal(t, len(d.Ports), len(cont.Ports))
```
[the remark this answers is not in the document]
Comment (BlaineEXE) [reply pkg/operator/ceph/test/container.go:92]: I tested it in the go playground, and it actually doesn't have a nil pointer exception.
https://play.golang.com/p/rgcuHPEb7rx

**Review by travisn — APPROVED** [review]

**Pull request merged**